### PR TITLE
refactor(function): make unit test goldenfile only display the used columns

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

### CODE REVIEW

File: src/query/functions-v2/tests/it/scalars/mod.rs [hunk 1]
```diff
@@ -117,6 +117,21 @@ pub fn run_ast(file: &mut impl Write, text: &str, columns: &[(&str, DataType, Co
                 Value::Column(output_col) => {
                     test_arrow_conversion(&output_col);
 
+                    // Only display the used input columns
+                    let used_columns = expr.column_refs();
+                    let input_domains = input_domains
```
Comment (sundy-li) [inline src/query/functions-v2/tests/it/scalars/mod.rs:122]: some functions should generate result column with row_size even without args.

For example:

`today()`, `random()`

So we should keep at least one column.


And we should refactor in  collect_columns function in https://github.com/datafuselabs/databend/blob/b3c2cb5b8a15e5f5c9c6f10f51edaa0ce162b145/src/query/functions-v2/tests/it/aggregates/mod.rs

Comment (andylokandy) [reply src/query/functions-v2/tests/it/scalars/mod.rs:122]: > some functions should generate result column with row_size even without args.

ok, let's do it in a separate pr. BTW, why `today()` needs the row size either?

**Review by b41sh — APPROVED** [review]

File: src/query/functions-v2/tests/it/scalars/mod.rs (hunk 1 above)
Comment (sundy-li) [reply src/query/functions-v2/tests/it/scalars/mod.rs:122]: sorry, `today` does not need.

**Review by sundy-li — APPROVED** [review]

File: src/query/functions-v2/tests/it/aggregates/mod.rs [hunk 2]
```diff
@@ -103,15 +104,15 @@ pub fn run_agg_ast(file: &mut impl Write, text: &str, columns: &[(&str, DataType
...
                         } else {
-                            vec![0]
+                            HashSet::from_iter([0])
                         }
                     }
-                    false => column_ids,
+                    false => used_columns,
                 };
 
                 for id in ids.iter() {
```
Comment (sundy-li) [inline src/query/functions-v2/tests/it/aggregates/mod.rs:118]: the iterator is  in arbitrary order, which makes the result unstable.

**Review by sundy-li — APPROVED** [review]

**Comment by b41sh** [comment]
`regexp.txt` also need to modify

**Pull request merged**